--- FoxySheepLexer.py.orig	2020-07-16 14:41:36.203587581 -0400
+++ FoxySheepLexer.py	2020-07-16 14:43:43.718369537 -0400
@@ -1,5 +1,15 @@
+# encoding: utf-8
+# In order for the generated lexer to subclass our lexer base class, we
+# have to patch the generated lexer using FoxySheepLexer.py.patch
+#
+# This is applied in the top-level Makefile. However to apply by hand run:
+#    patch < FoxySheepLexer.py.patch
+
+from antlr4.ParserRuleContext import RuleContext
+from FoxySheep.LexerBase import FoxySheepParser, LexerBase
+
 # Generated from FoxySheep.g4 by ANTLR 4.7.2
-from antlr4 import *
+from antlr4 import ATNDeserializer, DFA
 from io import StringIO
 from typing.io import TextIO
 import sys
@@ -534,7 +544,7 @@
         return buf.getvalue()
 
 
-class FoxySheepLexer(Lexer):
+class FoxySheepLexer(LexerBase):
 
     atn = ATNDeserializer().deserialize(serializedATN())
 
@@ -1002,6 +1012,3 @@
     def NEWLINE_action(self, localctx:RuleContext , actionIndex:int):
         if actionIndex == 23:
              self.checkNewline();
-     
-
-
