--- InputFormLexer.py.orig	2020-07-16 18:22:08.902288214 -0400
+++ InputFormLexer.py	2020-07-16 19:11:29.473668207 -0400
@@ -1,5 +1,16 @@
+# encoding: utf-8
+# In order for the generated lexer to subclass our lexer base class, we
+# have to patch the generated lexer using InputFormLexer.py.patch
+#
+# This is applied in the top-level Makefile. However to apply by hand run:
+#    patch < InputFormLexer.py.patch
+
+from antlr4.ParserRuleContext import RuleContext
+from FoxySheep.LexerBase import LexerBase
+from FoxySheep.generated.InputFormParser import InputFormParser, PredictionContextCache
+
 # Generated from InputForm.g4 by ANTLR 4.7.2
-from antlr4 import *
+from antlr4 import ATNDeserializer, DFA, LexerATNSimulator
 from io import StringIO
 from typing.io import TextIO
 import sys
@@ -534,7 +545,7 @@
         return buf.getvalue()
 
 
-class InputFormLexer(Lexer):
+class InputFormLexer(LexerBase):
 
     atn = ATNDeserializer().deserialize(serializedATN())
 
@@ -981,27 +992,24 @@
 
     def PLUS_action(self, localctx:RuleContext , actionIndex:int):
         if actionIndex == 19:
-             self.checkAdditiveOp(FoxySheepParser.BINARYPLUS); 
+             self.checkAdditiveOp(InputFormParser.BINARYPLUS);
      
 
     def MINUS_action(self, localctx:RuleContext , actionIndex:int):
         if actionIndex == 20:
-             self.checkAdditiveOp(FoxySheepParser.BINARYMINUS); 
+             self.checkAdditiveOp(InputFormParser.BINARYMINUS);
      
 
     def PLUSMINUS_action(self, localctx:RuleContext , actionIndex:int):
         if actionIndex == 21:
-             self.checkAdditiveOp(FoxySheepParser.BINARYPLUSMINUS); 
+             self.checkAdditiveOp(InputFormParser.BINARYPLUSMINUS);
      
 
     def MINUSPLUS_action(self, localctx:RuleContext , actionIndex:int):
         if actionIndex == 22:
-             self.checkAdditiveOp(FoxySheepParser.BINARYMINUSPLUS); 
+             self.checkAdditiveOp(InputFormParser.BINARYMINUSPLUS);
      
 
     def NEWLINE_action(self, localctx:RuleContext , actionIndex:int):
         if actionIndex == 23:
              self.checkNewline();
-     
-
-
